include/envoy/request_info/dynamic_metadata.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/common/exception.h"
+#include "envoy/common/pure.h"
+
+#include "common/common/fmt.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace RequestInfo {
+
+class DynamicMetadata {
+public:
+  class DynamicMetadataObject {
```

Comment:
Nit: If this is going to be nested in `DynamicMetadata`, you can just call it `Object`.

Reply:
Done.